exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/profile/ProfileWrapper.java

```diff
@@ -336,7 +336,8 @@ public String getOperatorsJSON() {
   }
 
   public Map<String, String> getQueryOptions() {
-    return getOptions(o -> OptionValue.OptionScope.QUERY == o.getScope());
+    // Skip reporting QUERY_MAX_ROWS if it is inapplicable and set to zero (e.g. query -> SHOW FILES)
+    return getOptions(o -> OptionValue.OptionScope.QUERY == o.getScope() && !(ExecConstants.QUERY_MAX_ROWS.equals(o.getName()) && String.valueOf(o.getValue()).equals("0")));
```

Comment:
Please revert this change.

Reply:
This change actually helps to not show the value of Zero being applied, when the query is inapplicable. Are you sure we want to remove this?

Reply:
I think we should not set max rows count to query context if it is the same as default, thus this count won't appear in Query Profile. Current solution just fixes the symptom not the real problem.

Reply:
Agreed. I am not sure, but I think I did this here 
https://github.com/apache/drill/blob/c3ee7949656fb4c1b144e1633f97002c159ec8f3/exec/java-exec/src/main/java/org/apache/drill/exec/ops/QueryContext.java#L120
because of the complexity in the logic  and (I think) because without this, the session value (0) gets precedence when the SYSTEM default exists.
I can revisit this post release to avoid blocking the release for now.

Reply:
Yes, as I wrote in the previous comment, we can skip the issue I mentioned in one of the comments for now.

Reply:
@kkhatua, you haven't reverted this change.